### DISCUSSION TITLE
Error Pulling Down Roster info from Baseball -- FIXED W UPDATE

### DISCUSSION THREAD

**Discussion by tomrulz** [cwendt94/espn-api#341] [ESPN API] [answered]
Hello. 

I have been using this api code for 2 seasons now (2021 and 2022). It has been working masterfully well and I am complete coding noob. When i run the script I have been using (and recently updated in april 2022, but has working daily since) the debugging gets stopped at the initialize league step with a key error 21. 

I would love to drop a file with my complete code but I am not sure I want to provide my league creds. 

Ill be totally honest, I haven't edited this code in any major way in about 18 months so its a little fuzzy at the moment. I figured I will ask first in case something simple has changed, otherwise I will spend time tomorrow trying to figure it out. 

Thank you in advance for the help!
tom

**Comment by cwendt94** [comment, marked as answer]
Hey Tom,

If you could just post the error message it will help narrow down where the error is happening. Most likely the package is expecting a value to be provided by ESPN but for some reason has changed.

**Reply by tomrulz** [reply]
Hey sorry I was out of town for the holiday. I have attached two files which should give you insight into the error.

**Reply by cwendt94** [reply]
No worries, that image showed the error happening when initializing the package but didn't specify the line number in the package.

When you try running it does it happen to show what file/line in the package it self?

**Reply by cwendt94** [reply]
I added some better error handling in the baseball package. If you would try updating to version 0.20.1 hopefully it fixed it.

**Reply by tomrulz** [reply]
ill check it out now! Thanks!

**Reply by tomrulz** [reply]
THAT FIXED IT!!!! I have no idea why but haha it works. Nice!!! Definitely still owe you some beers!

**Reply by cwendt94** [reply]
Nice! Glad it is fixed! It looks like a player has a position that is not defined in position mapping. So I just updated the code to handle unknown positions.

**Reply by tomrulz** [reply]
hmm I could probably investigate to see who that player is. I would assume its Ohtani but hes been in the league now for a while so unless they changed something with him its probably someone else.

**Reply by cwendt94** [reply]
It should be a player with a position as “21”. I think that was the key error you were seeing when trying to map position number to name.
